src/computability/turing_machine.lean

```diff
@@ -347,6 +781,7 @@ inductive stmt
   Both `Λ` and `Γ` are required to be inhabited; the default value
   for `Γ` is the "blank" tape value, and the default value of `Λ` is
   the initial state. -/
+@[nolint unused_arguments]
```

Comment:
A `nolint` attribute should always be accompanied by a written justification.  One common justification for `unused_arguments` is that we want different type class instances depending on the unused argument.  If this is the case here, please explain this.

Reply:
All the `nolint`s in the file were put there deliberately. (This is a small fraction of the number of unused arguments lints before the change.) In this case, the unused argument is a parameter, and the entire section is meant to be used together. The entire construction is in service of one theorem (that says that e.g. every TM1 program is TM0-computable), and so there is no need to get picky about the fact that the first half of the proof doesn't use the inhabited typeclass assumption.

Also in at least one case the "unused argument" is necessary in order to carry typeclass instances, as you say.

Reply:
Ah ok, that should be fine then.  Please add a comment like "unused parameters are due to parameters in this section, the definition is not used outside the section".